lib/remoteproc/rsc_table_parser.c

```diff
@@ -24,7 +24,7 @@ int handle_rsc_table(struct remoteproc *rproc,
 		     struct metal_io_region *io)
 {
 	char *rsc_start;
-	unsigned int rsc_type;
+	uint32_t rsc_type;
```

Comment:
I wonder here if we should not take oportunity o clean up the code using struct fw_rsc_hdr hdr